code/datums/flock/flock.dm

```diff
@@ -358,18 +360,56 @@
 	for (var/mob/M in src.annotation_viewers)
 		M.client?.images -= I
 
+// naming
+
+/datum/flock/proc/pick_name(flock_type)
+	var/name
+	var/name_found = FALSE
```

Comment:
You mean using break instead of name_found = TRUE?

Also, I don't think datums have the name var?

Reply:
Nevermind, datums don't have names. your right.
As for name_found, I'm saying that if you are using an associative list you don't need it, since the check "is this name used" is a single expression (`src.active_unit_names[name]`)

Reply:
I think the while() loop would break on first try if using !src.active_unit_names[name]?